src/libsystemd-network/dhcp6-option.c

```diff
@@ -581,7 +585,7 @@ int dhcp6_option_parse_ia(DHCP6Option *iaoption, DHCP6IA *ia) {
                 break;
         }
 
-        return 0;
+        return 1;
```

Comment:
we usually follow the rule that return paremeters are always initialized on success and never touched on failure. Given this is a success path, this should probably initialize `ret_status_code` to something, probably zero.

```c
if (ret_status_code)
        *ret_status_code = 0;

return 1;
```